mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -248,6 +269,11 @@ class KMeans @Since("1.5.0") (
       .setMaxIterations($(maxIter))
       .setSeed($(seed))
       .setEpsilon($(tol))
+
+    if (isSet(initialModel)) {
+      algo.setInitialModel($(initialModel).parentModel)
```

Comment:
1. MLlibKMeans doesn't check dimension match. If we want to check the dim, there are two ways, either use `rdd.first().size()` to get the dim, or we extract dim from attributes of DataFrame. The latter one is better but it need support from `Attribute`.

2. The `parentModel` of a `KMeansModel` is a `MLlibKMeansModel`, which supports cluster centers for `KMeans`. And `algo` is a `MLlibKMeans`. You can refer to the definition of `KMeanModel` in ML package. In fact, the name of `parentModel` is misleading.